packages/contracts-bedrock/test/L1/SystemConfig.t.sol

```diff
@@ -226,7 +228,140 @@ contract SystemConfig_Init_CustomGasToken is SystemConfig_Init {
         assertEq(systemConfig.gasPayingTokenSymbol(), token.symbol());
     }
 
-    function test_initialize_customGasToken_fails() external {
+    /// @dev Tests that initialization sets the correct values and getters work.
+    function testFuzz_initialize_customGasToken_succeeds(
+        address _token,
+        string calldata _name,
+        string calldata _symbol
+    )
+        external
+    {
+        vm.assume(bytes(_name).length <= 32);
+        vm.assume(bytes(_symbol).length <= 32);
+
+        // Wipe out the initialized slot so the proxy can be initialized again
+        vm.store(address(systemConfig), bytes32(0), bytes32(0));
+        vm.store(address(systemConfig), GasPayingToken.GAS_PAYING_TOKEN_SLOT, bytes32(0));
+        vm.store(address(systemConfig), GasPayingToken.GAS_PAYING_TOKEN_NAME_SLOT, bytes32(0));
+        vm.store(address(systemConfig), GasPayingToken.GAS_PAYING_TOKEN_SYMBOL_SLOT, bytes32(0));
+
+        vm.mockCall(_token, abi.encodeWithSelector(token.decimals.selector), abi.encode(18));
+        vm.mockCall(_token, abi.encodeWithSelector(token.name.selector), abi.encode(_name));
+        vm.mockCall(_token, abi.encodeWithSelector(token.symbol.selector), abi.encode(_symbol));
+
+        token = ERC20(_token);
+
+        vm.prank(systemConfig.owner());
```

Comment:
i am wondering why we are pranking as the owner, i technically don't think its necessary, it could be done in previous tests. we don't assert on the owner after init hmm but also pranking is only useful if `msg.sender` is touched in the init path